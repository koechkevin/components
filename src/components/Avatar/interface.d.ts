import { AvatarProps } from 'antd/lib/avatar';
import { CSSProperties, ReactNode } from 'react';

enum ChatStatus {
  online = 'online',
  offline = 'offline',
}

export interface Props extends AvatarProps {
  src?: string;
  dot?: boolean;
  status?: ChatStatus | string;
  size?: number | 'xsmall' | 'small' | 'medium' | 'large' | 'default';
  color?: string;
}
