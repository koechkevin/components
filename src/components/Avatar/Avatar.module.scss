@import '../../styles/global';

.avatar {
  &:global(.ant-avatar-square) {
    border-radius: 8px;
  }

  &:global(.ant-avatar img) {
    background-color: $color-basic-light-gray;
  }
}
