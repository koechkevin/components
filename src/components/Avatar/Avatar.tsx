import { Avatar as AntdAvatar } from 'antd';
import React, { FC } from 'react';

import styles from './Avatar.module.scss';

import { Props } from './interface.d';

export const Avatar: FC<Props> = (props) => {
  const { children, className, icon, src, shape, size, style, color } = props;
  const avatarSizes = {
    small: 32,
    medium: 40,
    large: 80,
    default: 'default',
  };
  const avatarSize = size && (avatarSizes[size] || size);
  const placeholderIcon = src ? icon : 'user';
  const classNames: string = [styles.avatar, className ? className : ''].join(' ');
  const avatarColor = color && !src ? { backgroundColor: color } : {};

  return (
    <AntdAvatar
      src={src}
      shape={shape}
      size={avatarSize}
      className={classNames}
      style={{ minWidth: avatarSize, ...avatarColor, ...style }}
      icon={!children ? placeholderIcon : ''}
    >
      {children}
    </AntdAvatar>
  );
};

Avatar.defaultProps = {
  size: 'default',
};

export default Avatar;
