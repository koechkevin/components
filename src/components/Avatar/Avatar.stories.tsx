import { number, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import { Col, Row, Typography } from 'antd';
import React from 'react';

import { Center } from '../Center';
import { ListItem } from '../SideBar/CompanyModal';
import Avatar from './Avatar';

import logo from '../../icons/logo.svg';
import mockLogo from '../../icons/mocks/company.png';

const src = 'https://s3.amazonaws.com/uifaces/faces/twitter/bowbrick/128.jpg';
const size = number('size', 200);
const initials = text('initials', 'AA');
const { Text } = Typography;
const companyList: ListItem[] = [
  { companyId: '123', avatarUrl: logo, name: 'Airbus', count: 4, avatarColor: '#fb6d77' },
  { companyId: '1234', avatarUrl: mockLogo, name: 'Pepsi', count: 21, avatarColor: '#46a094' },
  { companyId: '12345', avatarUrl: '', name: 'Airbus', avatarColor: '#5d85ce' },
];

storiesOf('Avatar', module)
  .addDecorator((story) => <Center>{story()}</Center>)
  .add('default', () => <Avatar />, { notes: 'Avatar default styles' })
  .add(
    'with circle shape and image',
    () => (
      <Row>
        <Row gutter={36}>
          <Col span={8}>
            <Text strong>24</Text> xsmall{' '}
          </Col>
          <Col span={8}>
            <Text strong>40</Text> small{' '}
          </Col>
          <Col span={8}>
            <Text strong>80</Text> medium{' '}
          </Col>
        </Row>
        <Row gutter={36}>
          <Col span={8}>
            <Avatar src={src} size="small" />
          </Col>
          <Col span={8}>
            <Avatar src={src} size="medium" />
          </Col>
          <Col span={8}>
            <Avatar src={src} size="large" />
          </Col>
        </Row>
      </Row>
    ),
    {
      notes: 'Avatar with Circle shape and image',
    },
  )
  .add(
    'with square shape and image',
    () => (
      <Row>
        <Row gutter={36}>
          <Col span={8}>
            <Text strong>24</Text> xsmall
          </Col>
          <Col span={8}>
            <Text strong>40</Text> small
          </Col>
          <Col span={8}>
            <Text strong>80</Text> medium
          </Col>
        </Row>
        <Row gutter={36}>
          <Col span={8}>
            <Avatar src={src} shape="square" size="small" />
          </Col>
          <Col span={8}>
            <Avatar src={src} shape="square" size="medium" />
          </Col>
          <Col span={8}>
            <Avatar src={src} shape="square" size="large" />
          </Col>
        </Row>
      </Row>
    ),
    {
      notes: 'Avatar with square shape and image',
    },
  )
  .add(
    'with square shape and initials',
    () => (
      <Row>
        <Row>
          <Col span={12}>
            <Text strong>32</Text> small
          </Col>
          <Col span={12}>
            <Text strong>40</Text> large
          </Col>
        </Row>
        <Row gutter={36}>
          <Col span={12}>
            <Avatar size="small" shape="square" color="#46a094" style={{ fontWeight: 'bold', fontSize: 10 }}>
              {initials}
            </Avatar>
          </Col>
          <Col span={12}>
            <Avatar size="medium" shape="square" color="#8260da" style={{ fontWeight: 'bold', fontSize: 20 }}>
              {initials}
            </Avatar>
          </Col>
        </Row>
      </Row>
    ),
    {
      notes: 'Avatar with Square shape and initials',
    },
  )
  .add(
    'with circle shape and initials',
    () => (
      <Row>
        <Row gutter={36}>
          <Col span={8}>
            <Text strong>32</Text> small
          </Col>
          <Col span={8}>
            <Text strong>40</Text> medium
          </Col>
          <Col span={8}>
            <Text strong>80</Text> large
          </Col>
        </Row>
        <Row gutter={36}>
          <Col span={8}>
            <Avatar size="small" shape="circle" color="#46a094" style={{ fontWeight: 'bold', fontSize: 10 }}>
              {initials}
            </Avatar>
          </Col>
          <Col span={8}>
            <Avatar size="medium" shape="circle" color="#8260da" style={{ fontWeight: 'bold', fontSize: 20 }}>
              {initials}
            </Avatar>
          </Col>
          <Col span={8}>
            <Avatar size="large" shape="circle" color="#8260da" style={{ fontWeight: 'bold', fontSize: 20 }}>
              {initials}
            </Avatar>
          </Col>
        </Row>
      </Row>
    ),
    {
      notes: 'Avatar with Square shape and initials',
    },
  )
  .add('with custom size', () => <Avatar src={src} size={size} />, {
    notes: 'Avatar with custom size',
  })
  .add('with an icon', () => <Avatar icon="user" />, {
    notes: 'Avatar with an icon',
  })
  .add('company logo', () => (
    <Row gutter={32}>
      {companyList.map((company) => (
        <Col span={8} key={company.companyId}>
          <Avatar src={company.avatarUrl} size="medium" shape="square" color={company.avatarColor}>
            {company.name[0]}
          </Avatar>
        </Col>
      ))}
    </Row>
  ));
