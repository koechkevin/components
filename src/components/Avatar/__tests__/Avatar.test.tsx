import Enzyme, { mount, shallow } from 'enzyme';
import React from 'react';
import Avatar from '../Avatar';

describe('component Avatar', () => {
  it('render without props ', () => {
    const tree = shallow(<Avatar />);
    expect(tree).toMatchSnapshot();
  });

  it('render with icon', () => {
    const tree = shallow(<Avatar icon="user" />);
    expect(tree).toMatchSnapshot();
  });

  it('render with size', () => {
    const size: string = 'size';
    const tree = shallow(<Avatar icon="user" />);
    expect(tree).toMatchSnapshot();
  });

  it('render with children', () => {
    const Image = (src: string) => (
      <img src={src} alt="avatar" style={{ width: '35px', height: '35px' }} />
    );

    const tree = shallow(
      <Avatar icon="user">
        <Image />
      </Avatar>
    );
    expect(tree).toMatchSnapshot();
  });

  it('className', () => {
    const fakeClassName = 'fakeClassName';
    const wrapper = shallow(<Avatar className={fakeClassName} />);
    expect(wrapper.hasClass('fakeClassName')).toEqual(true);
  });

  it('Avatar styling', () => {
    const fakeClassName = 'fakeClassName';
    const wrapper = shallow(<Avatar className={fakeClassName} />);
    const wrapperAvatar = wrapper.find('.fakeClassName');
    expect(wrapperAvatar.length).toEqual(1);
  });

  it('Avatar styling', () => {
    const style = { width: '30px' };
    const fakeClassName = 'fakeClassName';
    const wrapper = shallow(<Avatar className={fakeClassName} style={style} />);
    expect(wrapper.hasClass('fakeClassName')).toEqual(true);
    expect(wrapper.find('.fakeClassName')).toHaveLength(1);
  });
});
