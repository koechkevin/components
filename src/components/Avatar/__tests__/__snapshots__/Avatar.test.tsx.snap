// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Avatar render with children 1`] = `ShallowWrapper {}`;

exports[`component Avatar render with icon 1`] = `ShallowWrapper {}`;

exports[`component Avatar render with size 1`] = `ShallowWrapper {}`;

exports[`component Avatar render without props  1`] = `ShallowWrapper {}`;
